cirq/google/engine/engine_job.py

```diff
@@ -262,10 +262,15 @@ def results(self) -> List[study.TrialResult]:
         import cirq.google.engine.engine as engine_base
         if not self._results:
             job = self._refresh_job()
-            for _ in range(1000):
+            total_seconds_waited = 0.0
+            timeout = self.context.timeout
+            while True:
+                if timeout and total_seconds_waited >= timeout:
+                    break
                 if job.execution_status.state in TERMINAL_STATES:
                     break
                 time.sleep(0.5)
+                total_seconds_waited += 0.5
                 job = self._refresh_job()
```

Comment:
I guess this will not be exactly the timeout time since refresh job takes time.  But probably ok.